lib/widgets/component/floor_selection_dropdown.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:concordia_go/models/node.dart';
+import 'package:concordia_go/utilities/floor_maps_lib.dart';
+import 'package:flutter/material.dart';
+import 'package:concordia_go/utilities/application_constants.dart' as application_constants;
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:concordia_go/blocs/bloc.dart';
+
+class FloorSelectionDropdown extends StatefulWidget {
+  const FloorSelectionDropdown({Key key}) : super(key: key);
+
+  @override
+  FloorSelectionDropdownState createState() => FloorSelectionDropdownState();
+}
+
+class FloorSelectionDropdownState extends State<FloorSelectionDropdown> {
+  String dropdownValue;
+  String _buildingCode = 'H';
+  Map<String, List<Node>> _paths;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: const EdgeInsets.all(16.0),
+      child: Container(
+        padding: const EdgeInsets.all(3.0),
+        decoration: BoxDecoration(
+          color: Colors.white,
+          borderRadius: BorderRadius.circular(15.0),
+          border: Border.all(color: application_constants.concordiaRed, style: BorderStyle.solid, width: 1.80),
+        ),
+        child: BlocBuilder<MapBloc, MapState>(
+          builder: (BuildContext context, MapState state) {
+            if (state is IndoorMap) {
+              dropdownValue = state.floorLevel;
+              _buildingCode = state.buildingCode;
+              _paths = state.paths;
+            }
+            return DropdownButtonHideUnderline(
+              child: DropdownButton<String>(
+                value: dropdownValue,
+                icon: Icon(
+                  Icons.import_export,
+                  color: _paths != null && _paths[dropdownValue] == null
+                      ? Colors.grey
+                      : application_constants.concordiaRed,
+                ),
+                iconSize: 40,
```

Comment:
More random numbers

Reply:
Considering the size and the impact of the icon on the UI, it shouldn't cause any pixel overflow. I could use a constant size stored in a const variable, but there's no other comparable icon in that screen, so it would be the only property using that variable

Reply:
Alright sounds good, guess we won't worry about it since its not overflowing on any of our variety of phones